extensions/infinispan-embedded/runtime/src/main/java/io/quarkus/infinispan/embedded/runtime/InfinispanRecorder.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.infinispan.embedded.runtime;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class InfinispanRecorder {
+
+    public void configureRuntimeProperties(InfinispanEmbeddedRuntimeConfig infinispanEmbeddedRuntimeConfig) {
```

Comment:
hum does it mean the whole XML input is encoded at "built time" ? If I can't change the configuration at runtime, then why don't you pre-start all of the Infinispan engine, or at least pre-parse the configuration and already use this knowledge to e.g. not need any reflection at all?

Reply:
No, this is just passing the file name to be read at runtime to allow parsing of the XML file at runtime. I have tossed around the idea of having some flags to supply at build time to reduce some reflection classes, but haven't gotten that far yet. And the server already has a way to disable explicit classes to not add to the reflection list via `InfinispanReflectionExcludedBuildItem`.

Reply:
ah it's a filename. Ok that explains 90% of the doubts I had on this PR as I thought you want to benefit from the fact that "configuration is constant".

but why make the filename configurable? Shouldn't it just take "the one configuration" by convention?

Reply:
What do you mean by "one configuration" sorry?